### PR TITLE
[MM-30352] Automate backlogs - Plugins > Demo plugin functionality pt. 2 (2 test cases)

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

- [x] [MM-T2411](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2411) - Enable and disable plugin hook events for Demo plugin

- [x] [MM-T2406](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2406) - Plugin functions - Demo channel

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-30352

There are two cypress test cases in the linked ticket I have moved to the webapp repo, as they are not specific to the demo plugin:

- [MM-T2400](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2400) - Plugins Management

- [MM-T2391](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2391) - If plugins fail to start, they can be disabled

PR for these https://github.com/mattermost/mattermost-webapp/pull/7268

### CODE REVIEW

**Comment by DHaussermann** [comment]
@saturninoabril for this PR, and the PR from @jfrerich it is based on, the E2E folder is inside of webapp folder. Do you have any thoughts on this? Should we move the E2E folder to the top level?

Also, @mickmister has pointed out that Incident Response uses `tests-e2e` as the folder name in the top level of the repo. 

Can we confirm the standard we want to follow for this and other plugins such as just using `tests-e2e` as the folder name and having it sit at the top level?

**Comment by saturninoabril** [comment]
@DHaussermann Good point on standardization but I'll defer to the repo owner/maintainer. Honestly, I'd prefer to have it in the top-level folder since E2E is not only testing the webapp but the overall plugin application. I think the item to resolve is about the tooling like eslint, etc. Jason and I had some comments [here](https://github.com/mattermost/mattermost-plugin-demo/pull/130#discussion_r539711314) which resulted to have the folder inside the `/webapp` folder.

File: webapp/e2e/cypress/integration/demo_channel_spec.js
Comment (saturninoabril) [inline webapp/e2e/cypress/integration/demo_channel_spec.js:24]: Can safely remove.
```suggestion
```

File: webapp/e2e/cypress/integration/enable_disable_hooks_spec.js
Comment (saturninoabril) [inline webapp/e2e/cypress/integration/enable_disable_hooks_spec.js:23]: Same here.
```suggestion
```

File: webapp/e2e/cypress/integration/enable_disable_hooks_spec.js
Comment (saturninoabril) [inline webapp/e2e/cypress/integration/enable_disable_hooks_spec.js:26]: nit
```suggestion
        cy.apiRemovePluginById(pluginID);
```

File: webapp/e2e/cypress/integration/enable_disable_hooks_spec.js
Comment (saturninoabril) [inline webapp/e2e/cypress/integration/enable_disable_hooks_spec.js:33]:
```suggestion
        cy.apiRemovePluginById(pluginID);
```

**Review by saturninoabril — APPROVED** [review]
Thanks @mickmister! Tested and passed. LGTM. Left few comments.

**Pull request merged**
